Cannot install packages with Cygwin as source

### Checklist

- [X] I have verified this is the correct repository for opening this issue.
- [X] I have verified no other issues exist related to my problem.
- [X] I have verified this is not an issue for a specific package.
- [X] I have verified this issue is not security related.

### What You Are Seeing?

New versions of choco cannot install packages w/ `--source cygwin`. It looks like this:

![wtf](https://user-images.githubusercontent.com/355889/210188896-9f0618a5-bb94-4b3e-8c75-4c6b25c1997f.png)

Adding `--noop` shows that choco for some reason doesn't set the root installation directory in the command line:

![noop](https://user-images.githubusercontent.com/355889/210188947-2fe8f240-bee3-40bd-bcdc-7dfe091e32be.png)

### What is Expected?

Packages should be installed, obviously.

### How Did You Get This To Happen?

1. Install a new version of Chocolatey (1.2.0+).
2. Install Cygwin using `choco install cygwin -y --package-parameters='/InstallDir:C:\cg'`
3. Run `choco install cmake --source=cygwin`.

This __doesn't__ happen if after installing Chocolatey you downgrade it using `choco install chocolatey --version 1.1.0 --allow-downgrade -y`.

In both cases (with a new and an older version), I can see that choco reads HKLM\Software\Cygwin\setup\rootdir using `procmon`.

### System Details

* Operating System: 10.0.20348.0
* Windows PowerShell version: 5.1.20348.1366
* Chocolatey CLI Version: 1.2.1
* Chocolatey Licensed Extension version: 1.2.1
* Chocolatey License type:
* Terminal/Emulator: CMD

### Installed Packages

```bash
Chocolatey v1.2.1
chocolatey 1.2.1
chocolatey-compatibility.extension 1.0.0
chocolatey-core.extension 1.4.0
Cygwin 3.4.3
sysinternals 2022.11.28
5 packages installed.
```


### Output Log

```bash
2023-01-02 01:05:24,617 1156 [DEBUG] - XmlConfiguration is now operational
2023-01-02 01:05:24,680 1156 [DEBUG] - Adding new type 'WebPiService' for type 'ISourceRunner' from assembly 'choco'
2023-01-02 01:05:24,680 1156 [DEBUG] - Adding new type 'WindowsFeatureService' for type 'ISourceRunner' from assembly 'choco'
2023-01-02 01:05:24,680 1156 [DEBUG] - Adding new type 'CygwinService' for type 'ISourceRunner' from assembly 'choco'
2023-01-02 01:05:24,680 1156 [DEBUG] - Adding new type 'PythonService' for type 'ISourceRunner' from assembly 'choco'
2023-01-02 01:05:24,680 1156 [DEBUG] - Adding new type 'RubyGemsService' for type 'ISourceRunner' from assembly 'choco'
2023-01-02 01:05:24,680 1156 [DEBUG] - Adding new type 'SystemStateValidation' for type 'IValidation' from assembly 'choco'
2023-01-02 01:05:24,836 1156 [DEBUG] - Registering new command 'template' in assembly 'choco'
2023-01-02 01:05:24,836 1156 [DEBUG] - Registering new command 'upgrade' in assembly 'choco'
2023-01-02 01:05:24,836 1156 [DEBUG] - Registering new command 'export' in assembly 'choco'
2023-01-02 01:05:24,836 1156 [DEBUG] - Registering new command 'list' in assembly 'choco'
2023-01-02 01:05:24,836 1156 [DEBUG] - Registering new command 'info' in assembly 'choco'
2023-01-02 01:05:24,851 1156 [DEBUG] - Registering new command 'help' in assembly 'choco'
2023-01-02 01:05:24,851 1156 [DEBUG] - Registering new command 'config' in assembly 'choco'
2023-01-02 01:05:24,851 1156 [DEBUG] - Registering new command 'features' in assembly 'choco'
2023-01-02 01:05:24,851 1156 [DEBUG] - Registering new command 'new' in assembly 'choco'
2023-01-02 01:05:24,851 1156 [DEBUG] - Registering new command 'outdated' in assembly 'choco'
2023-01-02 01:05:24,851 1156 [DEBUG] - Registering new command 'pack' in assembly 'choco'
2023-01-02 01:05:24,851 1156 [DEBUG] - Registering new command 'pin' in assembly 'choco'
2023-01-02 01:05:24,883 1156 [DEBUG] - Registering new command 'push' in assembly 'choco'
2023-01-02 01:05:24,883 1156 [DEBUG] - Registering new command 'apikey' in assembly 'choco'
2023-01-02 01:05:24,883 1156 [DEBUG] - Registering new command 'source' in assembly 'choco'
2023-01-02 01:05:24,883 1156 [DEBUG] - Registering new command 'uninstall' in assembly 'choco'
2023-01-02 01:05:24,883 1156 [DEBUG] - Registering new command 'unpackself' in assembly 'choco'
2023-01-02 01:05:24,883 1156 [DEBUG] - Registering new command 'install' in assembly 'choco'
2023-01-02 01:05:25,023 1156 [INFO ] - ============================================================
2023-01-02 01:05:25,211 1156 [INFO ] - Chocolatey v1.2.1
2023-01-02 01:05:25,242 1156 [DEBUG] - Chocolatey is running on Windows v 10.0.20348.0
2023-01-02 01:05:25,258 1156 [DEBUG] - Attempting to delete file "C:/ProgramData/chocolatey/choco.exe.old".
2023-01-02 01:05:25,258 1156 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\choco.exe.old".
2023-01-02 01:05:25,289 1156 [DEBUG] - Command line: "C:\ProgramData\chocolatey\choco.exe" install cmake --source cygwin --debug --verbose
2023-01-02 01:05:25,305 1156 [DEBUG] - Received arguments: install cmake --source cygwin --debug --verbose
2023-01-02 01:05:25,367 1156 [DEBUG] - RemovePendingPackagesTask is now ready and waiting for PreRunMessage.
2023-01-02 01:05:25,398 1156 [DEBUG] - Sending message 'PreRunMessage' out if there are subscribers...
2023-01-02 01:05:25,414 1156 [DEBUG] - [Pending] Removing all pending packages that should not be considered installed...
2023-01-02 01:05:25,476 1156 [DEBUG] - Performing validation checks.
2023-01-02 01:05:25,492 1156 [DEBUG] - Global Configuration Validation Checks:
2023-01-02 01:05:25,492 1156 [DEBUG] -  - Package Exit Code / Exit On Reboot = Checked
2023-01-02 01:05:25,492 1156 [DEBUG] - System State Validation Checks:
2023-01-02 01:05:25,492 1156 [DEBUG] -  Reboot Requirement Checks:
2023-01-02 01:05:25,492 1156 [DEBUG] -  - Pending Computer Rename = Checked
2023-01-02 01:05:25,508 1156 [DEBUG] -  - Pending Component Based Servicing = Checked
2023-01-02 01:05:25,508 1156 [DEBUG] -  - Pending Windows Auto Update = Checked
2023-01-02 01:05:25,508 1156 [DEBUG] -  - Pending File Rename Operations = Ignored
2023-01-02 01:05:25,508 1156 [DEBUG] -  - Pending Windows Package Installer = Checked
2023-01-02 01:05:25,508 1156 [DEBUG] -  - Pending Windows Package Installer SysWow64 = Checked
2023-01-02 01:05:25,523 1156 [INFO ] - 2 validations performed. 2 success(es), 0 warning(s), and 0 error(s).
2023-01-02 01:05:25,539 1156 [DEBUG] - The source 'cygwin' evaluated to a 'cygwin' source type
2023-01-02 01:05:25,555 1156 [DEBUG] - 
NOTE: Hiding sensitive configuration data! Please double and triple
 check to be sure no sensitive data is shown, especially if copying
 output to a gist for review.
2023-01-02 01:05:25,570 1156 [DEBUG] - Configuration: CommandName='install'|
CacheLocation='C:\Users\Administrator\AppData\Local\Temp\2\chocolatey'|
ContainsLegacyPackageInstalls='True'|
CommandExecutionTimeoutSeconds='2700'|WebRequestTimeoutSeconds='30'|
Sources='cygwin'|SourceType='cygwin'|Debug='True'|Verbose='True'|
Trace='False'|Force='False'|Noop='False'|HelpRequested='False'|
UnsuccessfulParsing='False'|RegularOutput='True'|QuietOutput='False'|
PromptForConfirmation='True'|DisableCompatibilityChecks='False'|
AcceptLicense='False'|AllowUnofficialBuild='False'|Input='cmake'|
AllVersions='False'|SkipPackageInstallProvider='False'|
SkipHookScripts='False'|PackageNames='cmake'|Prerelease='False'|
ForceX86='False'|OverrideArguments='False'|NotSilent='False'|
ApplyPackageParametersToDependencies='False'|
ApplyInstallArgumentsToDependencies='False'|IgnoreDependencies='False'|
AllowMultipleVersions='False'|AllowDowngrade='False'|
ForceDependencies='False'|PinPackage='False'|
Information.PlatformType='Windows'|
Information.PlatformVersion='10.0.20348.0'|
Information.PlatformName='Windows Server 2016'|
Information.ChocolateyVersion='1.2.1.0'|
Information.ChocolateyProductVersion='1.2.1'|
Information.FullName='choco, Version=1.2.1.0, Culture=neutral, PublicKeyToken=79d02ea9cad655eb'|

Information.Is64BitOperatingSystem='True'|
Information.Is64BitProcess='True'|Information.IsInteractive='True'|
Information.UserName='Administrator'|
Information.UserDomainName='EC2AMAZ-VDKJE1B'|
Information.IsUserAdministrator='True'|
Information.IsUserSystemAccount='False'|
Information.IsUserRemoteDesktop='True'|Information.IsUserRemote='True'|
Information.IsProcessElevated='True'|
Information.IsLicensedVersion='False'|Information.LicenseType='Foss'|
Information.CurrentDirectory='C:\Users\Administrator'|
Features.AutoUninstaller='True'|Features.ChecksumFiles='True'|
Features.AllowEmptyChecksums='False'|
Features.AllowEmptyChecksumsSecure='True'|
Features.FailOnAutoUninstaller='False'|
Features.FailOnStandardError='False'|Features.UsePowerShellHost='True'|
Features.LogEnvironmentValues='False'|Features.LogWithoutColor='False'|
Features.VirusCheck='False'|
Features.FailOnInvalidOrMissingLicense='False'|
Features.IgnoreInvalidOptionsSwitches='True'|
Features.UsePackageExitCodes='True'|
Features.UseEnhancedExitCodes='False'|
Features.UseFipsCompliantChecksums='False'|
Features.ShowNonElevatedWarnings='True'|
Features.ShowDownloadProgress='True'|
Features.StopOnFirstPackageFailure='False'|
Features.UseRememberedArgumentsForUpgrades='False'|
Features.IgnoreUnfoundPackagesOnUpgradeOutdated='False'|
Features.SkipPackageUpgradesWhenNotInstalled='False'|
Features.RemovePackageInformationOnUninstall='False'|
Features.ExitOnRebootDetected='False'|
Features.LogValidationResultsOnWarnings='True'|
Features.UsePackageRepositoryOptimizations='True'|
ListCommand.LocalOnly='False'|ListCommand.IdOnly='False'|
ListCommand.IncludeRegistryPrograms='False'|ListCommand.PageSize='25'|
ListCommand.Exact='False'|ListCommand.ByIdOnly='False'|
ListCommand.ByTagOnly='False'|ListCommand.IdStartsWith='False'|
ListCommand.OrderByPopularity='False'|ListCommand.ApprovedOnly='False'|
ListCommand.DownloadCacheAvailable='False'|
ListCommand.NotBroken='False'|
ListCommand.IncludeVersionOverrides='False'|
UpgradeCommand.FailOnUnfound='False'|
UpgradeCommand.FailOnNotInstalled='False'|
UpgradeCommand.NotifyOnlyAvailableUpgrades='False'|
UpgradeCommand.ExcludePrerelease='False'|
NewCommand.AutomaticPackage='False'|
NewCommand.UseOriginalTemplate='False'|SourceCommand.Command='unknown'|
SourceCommand.Priority='0'|SourceCommand.BypassProxy='False'|
SourceCommand.AllowSelfService='False'|
SourceCommand.VisibleToAdminsOnly='False'|
FeatureCommand.Command='unknown'|ConfigCommand.Command='unknown'|
ApiKeyCommand.Remove='False'|PinCommand.Command='unknown'|
OutdatedCommand.IgnorePinned='False'|
ExportCommand.IncludeVersionNumbers='False'|Proxy.BypassOnLocal='True'|
TemplateCommand.Command='unknown'|
2023-01-02 01:05:25,586 1156 [DEBUG] - _ Chocolatey:ChocolateyInstallCommand - Normal Run Mode _
2023-01-02 01:05:25,586 1156 [DEBUG] - Running list with the following filter = ''
2023-01-02 01:05:25,601 1156 [DEBUG] - --- Start of List ---
2023-01-02 01:05:25,836 1156 [DEBUG] - chocolatey 1.2.1
2023-01-02 01:05:25,851 1156 [DEBUG] - chocolatey-compatibility.extension 1.0.0
2023-01-02 01:05:25,851 1156 [DEBUG] - chocolatey-core.extension 1.4.0
2023-01-02 01:05:25,867 1156 [DEBUG] - Cygwin 3.4.3
2023-01-02 01:05:25,867 1156 [INFO ] - Installing the following packages:
2023-01-02 01:05:25,867 1156 [INFO ] - cmake
2023-01-02 01:05:25,867 1156 [INFO ] - By installing, you accept licenses for the packages.
2023-01-02 01:05:25,914 1156 [DEBUG] - Calling command ['"cygwinsetup.exe" --quiet-mode --no-desktop --no-startmenu --root  --local-package-dir \packages --site http://mirrors.kernel.org/sourceware/cygwin/ --packages cmake']
2023-01-02 01:05:25,961 1156 [WARN ] - 
Chocolatey installed 0/0 packages. 
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
2023-01-02 01:05:25,961 1156 [DEBUG] - Sending message 'PostRunMessage' out if there are subscribers...
2023-01-02 01:05:25,976 1156 [ERROR] - Chocolatey had an error occur:
System.ComponentModel.Win32Exception (0x80004005): The system cannot find the file specified
   at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
   at chocolatey.infrastructure.commands.CommandExecutor.execute_static(String process, String arguments, Int32 waitForExitInSeconds, String workingDirectory, Action`2 stdOutAction, Action`2 stdErrAction, Boolean updateProcessPath, Boolean allowUseWindow, Boolean waitForExit)
   at chocolatey.infrastructure.commands.CommandExecutor.execute(String process, String arguments, Int32 waitForExitInSeconds, String workingDirectory, Action`2 stdOutAction, Action`2 stdErrAction, Boolean updateProcessPath, Boolean allowUseWindow)
   at chocolatey.infrastructure.app.services.CygwinService.install_run(ChocolateyConfiguration config, Action`1 continueAction)
   at chocolatey.infrastructure.app.services.ChocolateyPackageService.perform_source_runner_function[T](ChocolateyConfiguration config, Func`2 function)
   at chocolatey.infrastructure.app.services.ChocolateyPackageService.install_run(ChocolateyConfiguration config)
   at chocolatey.infrastructure.app.runners.GenericRunner.run(ChocolateyConfiguration config, Container container, Boolean isConsole, Action`1 parseArgs)
   at chocolatey.infrastructure.app.runners.ConsoleApplication.run(String[] args, ChocolateyConfiguration config, Container container)
   at chocolatey.console.Program.Main(String[] args)
2023-01-02 01:05:25,976 1156 [DEBUG] - Exiting with 1
```


### Additional Context

_No response_

--- a/src/chocolatey/infrastructure.app/services/CygwinService.cs
+++ b/src/chocolatey/infrastructure.app/services/CygwinService.cs
@@ -45,7 +45,20 @@ namespace chocolatey.infrastructure.app.services
         private const string PACKAGE_NAME_TOKEN = "{{packagename}}";
         private const string INSTALL_ROOT_TOKEN = "{{installroot}}";
         public const string CYGWIN_PACKAGE = "cygwin";
-        private string _rootDirectory = string.Empty;
+
+        private string _rootDirectory;
+        private string RootDirectory
+        {
+            get
+            {
+                if (string.IsNullOrWhiteSpace(_rootDirectory))
+                {
+                    _rootDirectory = get_root_directory();
+                }
+
+                return _rootDirectory;
+            }
+        }
 
         private const string APP_NAME = "Cygwin";
         public const string PACKAGE_NAME_GROUP = "PkgName";
@@ -156,8 +169,6 @@ namespace chocolatey.infrastructure.app.services
                 _nugetService.install_run(runnerConfig, ensureAction.Invoke);
                 config.PromptForConfirmation = prompt;
             }
-
-            set_root_dir_if_not_set();
         }
 
         public int count_run(ChocolateyConfiguration config)
@@ -167,21 +178,26 @@ namespace chocolatey.infrastructure.app.services
 
         public void set_root_dir_if_not_set()
         {
-            if (!string.IsNullOrWhiteSpace(_rootDirectory)) return;
+            if (!string.IsNullOrWhiteSpace(_rootDirectory))
+            {
+                return;
+            }
 
+            _rootDirectory = get_root_directory();
+        }
+
+        private string get_root_directory()
+        {
             var setupKey = _registryService.get_key(RegistryHive.LocalMachine, "SOFTWARE\\Cygwin\\setup");
             if (setupKey != null)
             {
-                _rootDirectory = setupKey.GetValue("rootdir", string.Empty).to_string();
+                return setupKey.GetValue("rootdir", string.Empty).to_string();
             }
 
-            if (string.IsNullOrWhiteSpace(_rootDirectory))
-            {
-                var binRoot = Environment.GetEnvironmentVariable("ChocolateyBinRoot");
-                if (string.IsNullOrWhiteSpace(binRoot)) binRoot = "c:\\tools";
+            var binRoot = Environment.GetEnvironmentVariable("ChocolateyBinRoot");
+            if (string.IsNullOrWhiteSpace(binRoot)) binRoot = "c:\\tools";
 
-                _rootDirectory = _fileSystem.combine_paths(binRoot, "cygwin");
-            }
+            return _fileSystem.combine_paths(binRoot, "cygwin");
         }
 
         public string get_exe(string rootpath)
@@ -203,7 +219,7 @@ namespace chocolatey.infrastructure.app.services
         {
             var args = ExternalCommandArgsBuilder.build_arguments(config, argsDictionary);
 
-            args = args.Replace(INSTALL_ROOT_TOKEN, _rootDirectory);
+            args = args.Replace(INSTALL_ROOT_TOKEN, RootDirectory);
 
             return args;
         }
@@ -211,7 +227,7 @@ namespace chocolatey.infrastructure.app.services
         public void install_noop(ChocolateyConfiguration config, Action<PackageResult> continueAction)
         {
             var args = build_args(config, _installArguments);
-            this.Log().Info("Would have run '{0} {1}'".format_with(get_exe(_rootDirectory).escape_curly_braces(), args.escape_curly_braces()));
+            this.Log().Info("Would have run '{0} {1}'".format_with(get_exe(RootDirectory).escape_curly_braces(), args.escape_curly_braces()));
         }
 
         public ConcurrentDictionary<string, PackageResult> install_run(ChocolateyConfiguration config, Action<PackageResult> continueAction)
@@ -224,7 +240,7 @@ namespace chocolatey.infrastructure.app.services
                 var argsForPackage = args.Replace(PACKAGE_NAME_TOKEN, packageToInstall);
 
                 var exitCode = _commandExecutor.execute(
-                    get_exe(_rootDirectory),
+                    get_exe(RootDirectory),
                     argsForPackage,
                     config.CommandExecutionTimeoutSeconds,
                     _fileSystem.get_current_directory(),

--- a/src/chocolatey/infrastructure.app/services/CygwinService.cs
+++ b/src/chocolatey/infrastructure.app/services/CygwinService.cs
@@ -176,6 +176,7 @@ namespace chocolatey.infrastructure.app.services
             throw new NotImplementedException("Count is not supported for this source runner.");
         }
 
+        [Obsolete("This method does not need to be called publicly and may be made private or removed in a later version.")]
         public void set_root_dir_if_not_set()
         {
             if (!string.IsNullOrWhiteSpace(_rootDirectory))
@@ -200,6 +201,7 @@ namespace chocolatey.infrastructure.app.services
             return _fileSystem.combine_paths(binRoot, "cygwin");
         }
 
+        [Obsolete("This method does not need to be called publicly and may be made private or removed in a later version.")]
         public string get_exe(string rootpath)
         {
             return _fileSystem.combine_paths(rootpath, "cygwinsetup.exe");
@@ -215,6 +217,7 @@ namespace chocolatey.infrastructure.app.services
             throw new NotImplementedException("{0} does not implement list".format_with(APP_NAME));
         }
 
+        [Obsolete("This method does not need to be called publicly and may be made private or removed in a later version.")]
         public string build_args(ChocolateyConfiguration config, IDictionary<string, ExternalCommandArgument> argsDictionary)
         {
             var args = ExternalCommandArgsBuilder.build_arguments(config, argsDictionary);

